public/app/plugins/panel/logs/useNewData.tsx

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from 'react';
+import { cloneDeep } from 'lodash';
+
+interface Props {
+  data: any;
+}
+
+const useNewData = ({ data }: Props) => {
```

Comment:
@MikeFuster I'd change the name of this hook to something like usePopulatedData 💪

Reply:
Done!